### ISSUE TITLE
host protection username/login

### ISSUE THREAD

**Issue by cs35-owncloud** [miroslavpejic85/mirotalksfu#161]
# Bug Report

## Describe the bug

When enabling host protection, you need to type in again your username when you create a room and try to join.
It's not really a bug per se, but it's quite annoying to have to type again your username.
Moreover, before the introduction of allowed_rooms, one could set any name.
It would be nice to have a login and a username for example:

```
    host: {
        protected: true,
        user_auth: false,,
        users: [
            {
                login: 'test',
                username: 'Jean Dupons',
                password: 'test',
                allowed_rooms: ['*'],
            },
        ],
    },

```
This way, you have to login at first with your login, here being test, and once you are logged in, mirotalk takes the username as a first option for the name in the rooms.
It's way easier for humans to have a username rather a login when in a room.

## To reproduce

1. Enable host protection:

```
    host: {
        protected: true,
        user_auth: false,
        users: [
            {
                username: 'test',
                password: 'test',
                allowed_rooms: ['*'],
            },
        ],
    },

```

3. Log in to your instance
![mirotalksfu_login](https://github.com/user-attachments/assets/5b37f53e-8888-4010-8fe6-70c5ef86f3bc)

4. Create a room and in the section "Enter your email or name", type something different than your username

![mirotalksfu_meeting](https://github.com/user-attachments/assets/b94d818c-96e9-4258-98a1-e851f80af2e6)

5. You end up having an error of room not allowed even though you are logged in

![mirotalksfu_meeting_error](https://github.com/user-attachments/assets/9a22360d-93b5-46f6-905b-79024eecf7cc)

## Expected behavior

Before the introduction of allowed_rooms, any logged in user could use a different and more human readable username.

## Screenshots

See above

## Desktop - Mobile

Please complete the following information:

 - Device or OS: Linux Mint
 - Browser : Firefox

## Additional context

None

Thanks !

**Comment by miroslavpejic85** [comment]
Update your `app/src/config.js` by adding optional `displayname`: 

```js
const config = {
    host: {
        protected: true,
        user_auth: false,
        users: [
            {
                username: 'test',
                password: 'test',
                displayname: 'Jean Dupont', // Optional: Set the display name for meetings
                allowed_rooms: ['*'],
            },
        ],
    },
};
```

---

![mirotalk-sfu-displayname-feature](https://github.com/user-attachments/assets/5c2341e2-9801-45bb-b33e-b64ea3beb847)

**Comment by cs35-owncloud** [comment]
Hello,

thanks again for your quick response and quick fix !
I pulled the latest version but as of now it doesn't seem to work, I guess it's not up to date yet in the docker repository ?

```
  app_version: '1.5.50',
  node_version: '20.16.0',
  ```

Thanks again !

**Comment by miroslavpejic85** [comment]
> thanks again for your quick response and quick fix !

You're welcome.

> I pulled the latest version but as of now it doesn't seem to work, I guess it's not up to date yet in the docker repository ?

## **Update and Deployment Instructions**

1. **Update `config.js`**:
   - Make the required changes to your `config.js` file as described. Ensure that the `displayname` field is correctly implemented for each desired users.

2. **Wait for the Build Process to Complete**:
   - Monitor the build progress using the following links:
     - [Build Run #1](https://github.com/miroslavpejic85/mirotalksfu/actions/runs/10349985697/job/28645498455)
     - [Build Run #2](https://github.com/miroslavpejic85/mirotalksfu/actions/runs/10350161876/job/28646017313)

3. **Pull the Updated Image and Restart Your Instance**:
   - Once the build process is complete, pull the updated Docker image and restart your instance to apply the new configuration.

**Comment by cs35-owncloud** [comment]
Thanks, my mistake for being impatient, I'll monitor the build, thanks again for your work !

**Comment by miroslavpejic85** [comment]
> Thanks, my mistake for being impatient, I'll monitor the build, thanks again for your work !

You're welcome, there is the alternative to speedup the tests :)

### `docker-compose.yaml`:

```yaml
...
volumes:
     - ./app/:/src/app/:ro
     - ./public/:/src/public/:ro
...
```

- **Volumes:** Maps `./app` and `./public` from your host to `/src/app` and `/src/public` in the container as read-only. This means:
  - **Immediate Updates:** Changes in your local directories reflect instantly in the container.
  - **Read-Only:** Prevents the container from modifying these files.

### Commands:

1. **`git pull`**: Updates your local code with the latest changes.
2. **`docker-compose down`**: Stops and removes the current containers.
3. **`docker-compose up -d`**: Restarts containers with the updated code and configuration in detached mode.

This setup speeds up testing by keeping the container in sync with local changes without rebuilding/waiting the new image.

**Comment by cs35-owncloud** [comment]
Alright, it's built, I just tried, and it allows now the displayname with is fine ! Thanks.
The only "downside" as of now, is that it does not load my displayname once I logged in with my username unlike in your video recording.

**Comment by miroslavpejic85** [comment]
> Alright, it's built, I just tried, and it allows now the displayname with is fine ! Thanks.

Thank you for confirming that the display name issue has been resolved.

Please note that the name might be cached based on the RoomId. If you encounter any inconsistencies, consider clearing your cache and cookies.

I will go ahead and close this task. Thank you for your cooperation!
